File: Source/Samples/95_Periodic/code_ascrollcontainer.cpp
// AScrollContainer application source code
#include "PeriodicApp.h"

void PeriodicApp::setup_ascrollcontainer( AWidget *layout )
{
    PODVector<AWidget*> dest;
    layout->SearchWidgetClass( "TBButton", dest );
    for (unsigned ii = 0; ii < dest.Size(); ii++)
        SubscribeToEvent(dest[ii], E_WIDGETEVENT, URHO3D_HANDLER(PeriodicApp, HandleAscrollcontainerEvent ));
}

void PeriodicApp::HandleAscrollcontainerEvent(StringHash eventType, VariantMap& eventData)
{
    using namespace WidgetEvent;
    AWidget* widget = static_cast<AWidget*>(eventData[P_TARGET].GetPtr());
    if ( widget == NULL ) return;
    if (eventData[P_TYPE] == UI_EVENT_TYPE_CLICK)
    {
        if (widget->GetId() == "uiscrollcontainercode" )
        {
            AppLog( " support : " + widget->GetId() + " was pressed " );
            ViewCode ( "Sources/code_ascrollcontainer.cpp", widget->GetParent() );
        }
        if (widget->GetId() ==  "uiscrollcontainerlayout" )
        {
            AppLog( " support : " + widget->GetId() + " was pressed ");
            ViewCode ( "Scenes/layout_ascrollcontainer.ui.txt", widget->GetParent() );
        }
    }
}

File: bin/Data/AUI/Periodic/LuaScripts/code_ascrollcontainer.lua
-- AScrollContainer application source code

function setup_uiscrollcontainer( layout )
    local dest = layout:SearchAWidgetClass( "TBButton" )
    for i, awid in ipairs(dest) do
        SubscribeToEvent(awid, "WidgetEvent", "HandleUiscrollcontainerEvent")
	end
end

function HandleUiscrollcontainerEvent( eventType, eventData)
    local widget = eventData["Target"]:GetPtr("AWidget")
    if widget == nil then
 		return
	end
    if eventData["Type"]:GetInt() == UI_EVENT_TYPE_CLICK then
        if widget:GetId() == "uiscrollcontainercode" then
            AppLog( " support : " .. widget:GetId() .. " was pressed " )
            ViewCode ( "LuaScripts/code_ascrollcontainer.lua", widget:GetParent())
        end
        if widget:GetId() ==  "uiscrollcontainerlayout" then
            AppLog( " support : " .. widget:GetId() .. " was pressed ")
            ViewCode ( "Scenes/layout_ascrollcontainer.ui.txt", widget:GetParent())
        end
    end
end

